packages/synapse-react-client/src/utils/functions/RegularExpressions.test.ts

```diff
@@ -0,0 +1,14 @@
+import { parseSynId } from './RegularExpressions'
+
+describe('RegularExpressions', () => {
+  describe('parseSynId', () => {
+    test.each([
+      ['syn123', { targetId: 'syn123', targetVersionNumber: undefined }],
+      ['syn123.4', { targetId: 'syn123', targetVersionNumber: 4 }],
+      ['Syn123.4', { targetId: 'syn123', targetVersionNumber: 4 }],
+      ['123.4', null],
```

Comment:
For `parseSynId`, I am explicitly not handling the case with no syn prefix because that's the current behavior.